tests/Functional/Driver/OCI8/ConnectionTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\OCI8;
+
+use Doctrine\DBAL\DriverManager;
+use Doctrine\DBAL\Exception\DriverException;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+use PHPUnit\Framework\Attributes\RequiresPhpExtension;
+use Throwable;
+
+#[RequiresPhpExtension('oci8')]
+class ConnectionTest extends FunctionalTestCase
+{
+    private const DISCONNECT_SESSION_SQL = <<<'SQL'
+        DECLARE
+            v_sid    VARCHAR2(10) := :sid;
+            v_serial VARCHAR2(10) := :serial;
+            v_sql    VARCHAR2(200);
+        BEGIN
+            v_sql := 'ALTER SYSTEM DISCONNECT SESSION ''' || v_sid || ',' || v_serial || ''' IMMEDIATE';
+            EXECUTE IMMEDIATE v_sql;
+        END;
+SQL;
+
+    public function testPrepareThrowsErrorOnConnectionLost(): void
+    {
+        $this->markConnectionNotReusable();
+        $this->killCurrentSession();
+
+        $this->expectException(DriverException::class);
+
+        $this->connection->prepare('SELECT * FROM 1');
+    }
+
+    /**
+     * Get the session details
+     *
+     * @return array<string, mixed>|false
+     */
+    private function getCurrentSession(): array|false
+    {
+        $stmt = $this->connection->prepare("
+            SELECT SID, SERIAL#
+            FROM V\$SESSION
+            WHERE AUDSID = USERENV('SESSIONID')
+        ");
+
+        return $stmt->executeQuery()->fetchAssociative();
+    }
+
+    /**
+     * Kill the current session, by using another connection
+     * Oracle doesn't allow you to terminate the current session, so we use a second connection
+     */
+    private function killCurrentSession(): void
+    {
+        $session = $this->getCurrentSession();
+        if ($session === false || ! isset($session['SID'], $session['SERIAL#'])) {
+            return;
+        }
+
+        $params            = TestUtil::getConnectionParams();
+        $params['charset'] = 'utf8mb4';
+        $secondConnection  = DriverManager::getConnection($params);
+
+        try {
+            $stmt = $secondConnection->prepare(self::DISCONNECT_SESSION_SQL);
+            $stmt->bindValue(':sid', $session['SID']);
+            $stmt->bindValue(':serial', $session['SERIAL#']);
+            $stmt->executeQuery();
+        } catch (Throwable $e) {
+            self::fail($e->getMessage());
```

Comment:
This try-catch block can be removed. If the query fails, it will automatically fail the test.